kedro/extras/datasets/api/api_dataset.py

```diff
@@ -57,15 +58,29 @@ def __init__(
             headers: The HTTP headers.
                 https://requests.readthedocs.io/en/master/user/quickstart/#custom-headers
             auth: Anything ``requests`` accepts. Normally it's either ``('login', 'password')``,
-                or ``AuthBase``, ``HTTPBasicAuth`` instance for more complex cases.
+                or ``AuthBase``, ``HTTPBasicAuth`` instance for more complex cases. Any
+                iterable will be cast to a tuple.
             json: The request payload, used for POST, PUT, etc requests, passed in
                 to the json kwarg in the requests object.
                 https://requests.readthedocs.io/en/master/user/quickstart/#more-complicated-post-requests
             timeout: The wait time in seconds for a response, defaults to 1 minute.
                 https://requests.readthedocs.io/en/master/user/quickstart/#timeouts
+            credentials: same as ``auth``. Allows specifying ``auth`` secrets in
+                credentials.yml.
 
+        Raises:
+            ValueError: if both ``credentials`` and ``auth`` are specified.
         """
         super().__init__()
+
+        if credentials is not None and auth is not None:
+            raise ValueError("Cannot specify both auth and credentials.")
+
+        auth = credentials or auth
+
+        if isinstance(auth, Iterable):
```

Comment:
I did not know you could do this!

Reply:
So it works!
![image](https://user-images.githubusercontent.com/35801847/145248343-6f181e96-8b93-4e2b-849f-eb979962b9ca.png)

Regular strings obv cause issues, but they don't meet the requirements of the type hint